docs/interactions/checks.rst

```diff
@@ -0,0 +1,182 @@
+
+Custom Check
+-------------
+
+Custom check commands can also be implemented if further functionality is needed outside of the checks listed above. These will generally come in two forms: a standard check decorator that is passed a custom function, or a custom decorator for common checks.
+
+.. code-block:: python3
+    :emphasize-lines: 6,17
+
+    # Custom check function.
+    def check_if_owner(interaction: discord.Interaction) -> bool:
+        return interaction.guild is not None and interaction.user.id == interaction.guild.owner_id
+
+    @app_commands.command()
+    @app_commands.check(check_if_owner)
+    async def owners_only(interaction: discord.Interaction):
+        await interaction.response.send_message('Hello boss!')
+
+    # Custom check decorator.
+    def is_me():
+        def predicate(interaction: discord.Interaction) -> bool:
+            return interaction.user.id == 1234567890
+        return app_commands.check(predicate)
+
+    @app_commands.command()
+    @is_me()
+    async def me_only(interaction: discord.Interaction):
+        await interaction.response.send_message('I know you!', ephemeral=True)
+
+In this example, the first check is implemented to only allow the command usage if the user who runs the command is the owner of the guild the command is used in. This is simply a function which takes a :class:`.Interaction` as an argument, which is passed to a :func:`@check <.app_commands.check>` decorator. The second check implemented creates a custom decorator which checks if the user who activates a command has a specific id. This limits a command to only one user.
+
+Combining Checks
+-----------------
+
+You can combine multiple checks on a command to end up with a very particular functionality for your needs, in a very concise manner.
+
+As an example, here is a command which only runs if a user has a certain role, ``send_messages`` permissions, the bot has ``manage_messages`` permissions, and with a ``15`` second global cooldown:
+
+.. code-block:: python3
+
+    @app_commands.command()
+    @app_commands.checks.has_role('Necessary Role')
+    @app_commands.checks.has_permissions(send_messages=True)
+    @app_commands.checks.bot_has_permissions(manage_messages=True)
+    @app_commands.checks.cooldown(1, 15.0, key=None)
+    async def check_example(interaction: discord.Interaction):
+        await interaction.send_message('Checks passed!', ephemeral=True)
+
+.. note::
+
+    When checks are failed, they will throw an error. By implementing custom error handling for these errors, you can create a system that will do something different when a check is failed. See the :ref:`error handling guide <guide_interactions_error-handling>` for further information on this.
```

Comment:
> When checks are failed

This sounds odd, maybe something like `When checks fail` or `When checks do not pass`